src/electionguard_gui/services/ballot_upload_service.py

```diff
@@ -0,0 +1,55 @@
+from datetime import datetime
+from bson import ObjectId
+from pymongo.database import Database
+from electionguard_gui.services.eel_log_service import EelLogService
+from electionguard_gui.services.service_base import ServiceBase
+from electionguard_gui.services.authorization_service import AuthorizationService
+
+
+class BallotUploadService(ServiceBase):
+    """Responsible for functionality related to ballot uploads"""
+
+    _log: EelLogService
+    _auth_service: AuthorizationService
+
+    def __init__(
+        self, log_service: EelLogService, auth_service: AuthorizationService
+    ) -> None:
+        self._log = log_service
+        self._auth_service = auth_service
+
+    def create(
+        self,
+        db: Database,
+        election_id: str,
+        device_file_name: str,
+        device_file_contents: str,
+    ) -> str:
+        election = {
+            "election_id": election_id,
+            "device_file_name": device_file_name,
+            "device_file_contents": device_file_contents,
+            "ballots": [],
+            "created_by": self._auth_service.get_user_id(),
+            "created_at": datetime.utcnow(),
+        }
+        self._log.trace(f"inserting ballot upload for: {election_id}")
+        inserted_id = db.ballot_uploads.insert_one(election).inserted_id
```

Comment:
One thing I struggle with in NoSQL databases is when to use a new collection vs how much stuff to stuff into an existing record of an existing collection.  This felt like a new collection with a "foreign key" kind of situation, otherwise the election record would just get gigantic with all those ballots in it.

Reply:
The other option is a ballot record type inside of the election record collection.  This way you can query for just one type or the other and it is all inside of a single collection.